plugins/handler/plugin_impl_handler.go

```diff
@@ -66,55 +82,40 @@ func (p *Plugin) Init() error {
 	return nil
 }
 
-// AfterInit is called for post init processing
-func (p *Plugin) AfterInit() error {
-	p.Log.Info("AfterInit")
-
-	ticker := time.NewTicker(objectstore.ObjectStoreReadyInterval)
-	timeout := time.After(objectstore.ObjectStoreReadyTimeout)
-	defer ticker.Stop()
-	// Wait for objectstore to initialize
-	ready := false
-	for !ready {
-		select {
-		case <-timeout:
-			return fmt.Errorf("timeout waiting for ObjectStore")
-		case <-ticker.C:
-			if p.objectStore = objectstore.SharedPlugin(); p.objectStore != nil {
-				ready = true
-				ticker.Stop()
-				p.Log.Info("ObjectStore is ready, starting Consumer")
-			} else {
-				p.Log.Info("ObjectStore is not ready, waiting")
-			}
-		}
-	}
-
-	return nil
-}
-
 // Close is called when the plugin is being stopped
 func (p *Plugin) Close() error {
+	return p.IdempotentClose(p.close)
+}
+
+func (p *Plugin) close() error {
 	p.Log.Info("Close")
+	err := p.Log.Close()
+	if err != nil {
+		return err
+	}
+	err = p.ObjectStore.Close()
+	if err != nil {
+		return err
+	}
 
-	return nil
+	return err
 }
 
 // ObjectCreated is called when an object is created
 func (p *Plugin) ObjectCreated(obj interface{}) {
 	p.Log.Infof("LogCrdHandler.ObjectCreated: ", reflect.TypeOf(obj), obj)
-	p.objectStore.ObjectCreated(obj)
+	p.ObjectStore.ObjectCreated(obj)
```

Comment:
I think ObjectStore should stay private.

Reply:
If its private, nobody can wire in an alternative implementation.  If that's what we want, it can be done that way.  The big question is whether we want to weld ourselves to a particular ObjectStore implementation.

Reply:
ObjectStore is not really a plugin, it is strictly internal thing used in deep conjunction with crd plugin and I doubt it needs to be swapped with an alternative implementation. The goal of making it private was to protect data, as now once it public the data can be accessed in a non control way and since it serves to multiple plugins data consistency is a big question. In methods we expose we control access with mutexes, hence guaranteeing data consistency.

Reply:
So... the Dep is stored using the ObjectStore interface, so if someone wants to much with it, the only things they can do are:

a) Replace it with an alternate ObjectStore (which they can always do with UseDeps() anyway) or
b) Use the ObjectStore interface

If they want to do #b, then they could just call objectstore,SharedPlugin() and get the ObjectStore that way and muck with it.

I don't quite see how this increases exposure...

Reply:
@edwarnicke Once objectStore becomes ObjectStore, all data structures within this type becomes visible, so the consumer will see them and can use them without bother of using exposed interfaces.

Reply:
@sbezverk Anyone can call SharedPlugin() and get the ObjectStore... they don’t need to get it from Deps... how does not treating it as a proper Dep make it less accessible?